pylutron/__init__.py

```diff
@@ -87,6 +87,7 @@ def send(self, cmd):
 
     Must not hold self._lock.
     """
+    self._maybe_reconnect()
```

Comment:
I'll have to page back in the context of connection maintenance, but this breaks the assumption stated in block comment on line 135. The _maybe_reconnect() is only meant to run from the main loop and I'll have to think about implications if it doesn't.

I think a simpler fix (and one that would simply ignore sends when connection is dead) is to make send_locked first check if we are connected before trying to send. 

Could you try that instead?

Reply:
Going over the code, I think the problem is that when we notice a connection problem on send, we propagate one of several possible exceptions back to home-assistant, which doesn't really handle this well at all, and for reasons I don't understand, seems to persist in a broken state.

I suppose I could attempt to debug and diagnose the interactions here more by attempting to reproduce the issue in my own environment.

Reply:
I've been stuck on 0.84.6 since January and am really eager to see some progress on pylutron. Wish I had the knowledge to actively contribute -- apologies -- but am happy to serve as a tester. Feel free to call on me.

Reply:
Ok, yeah, I'm working on a better fix for this. Lots of things don't like when connections die. telnetlib is not great about it.